src/internal.c

```diff
@@ -28958,13 +28968,20 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
                 WOLFSSL_MSG("Prev session's cipher suite not in ClientHello");
             #ifdef WOLFSSL_EXTRA_ALERTS
                 SendAlert(ssl, alert_fatal, illegal_parameter);
+            #endif
+            #ifdef HAVE_EXT_CACHE
+                if (gotSess) {
+                    wolfSSL_SESSION_free(session);
+                }
             #endif
                 return UNSUPPORTED_SUITE;
             }
         #endif
 
         #ifdef HAVE_EXT_CACHE
-            wolfSSL_SESSION_free(session);
+            if (gotSess) {
```

Comment:
Session could be freed at end of if statement.
The last else branch can free after loop and set session to NULL.